package/Shaders/ISVolumetricLightingGenerateCS.hlsl

```diff
@@ -33,6 +34,69 @@ cbuffer PerTechnique : register(b0)
+ * @param g The anisotropy factor (-1 for full backward scattering, 1 for full forward scattering, and 0 for isotropic).
+ * @return The Schlick phase function value for the given parameters.
+ *
+ * @note Performance vs Quality:
+ * - Schlick is a faster approximation compared to Henyey-Greenstein because it avoids the expensive power operations.
+ * - It provides visually similar results but sacrifices some accuracy, especially for complex anisotropic scattering.
+ * - Ideal for real-time rendering where performance is critical, such as games (default for Skyrim).
+ */
+float SchlickPhase(float cosTheta, float g)
+{
+	float k = (1 - g * g) / (4 * M_PI * pow(1 + g * cosTheta, 2));
+	return k;
+}
+
+/**
+ * @brief Computes the Henyey-Greenstein phase function for scattering.
+ *
+ * The Henyey-Greenstein phase function is a widely used model for simulating light scattering in participating media.
+ * It accurately models the anisotropy of light scattering, providing good realism for both forward and backward scattering.
+ * 
+ * @param cosTheta The cosine of the scattering angle (cosine of the angle between the light and the view direction).
+ * @param g The anisotropy factor (-1 for full backward scattering, 1 for full forward scattering, and 0 for isotropic).
+ * @return The Henyey-Greenstein phase function value for the given parameters.
+ *
+ * @note Performance vs Quality:
+ * - More computationally expensive than Schlick due to the use of power operations.
+ * - Offers a more physically accurate representation of scattering, particularly for media with pronounced forward or backward scattering behavior.
+ * - Suitable for applications where visual fidelity is more important than real-time performance.
+ */
+float HenyeyGreensteinPhase(float cosTheta, float g)
+{
+	float g2 = g * g;
+	return (1.0 - g2) / (4.0 * M_PI * pow(1.0 + g2 - 2.0 * g * cosTheta, 1.5));
+}
+
+/**
+ * @brief Computes the Mie phase function for light scattering.
+ *
+ * The Mie phase function models the scattering of light by larger particles, such as water droplets or fog.
+ * It is more complex than the Henyey-Greenstein phase function and is often used to model more realistic atmospheric effects.
+ * 
+ * @param cosTheta The cosine of the scattering angle (cosine of the angle between the light and the view direction).
+ * @param g The anisotropy factor (-1 for full backward scattering, 1 for full forward scattering, and 0 for isotropic).
+ * @return The Mie phase function value for the given parameters.
+ *
+ * @note Performance vs Quality:
+ * - More computationally expensive than both Schlick and Henyey-Greenstein due to the complex denominator and power operations.
+ * - Produces the most physically accurate results, especially for scenes involving larger particles such as clouds or fog.
+ * - Best suited for high-fidelity rendering applications, but might be too slow for real-time gaming without optimization.
+ */
+float MiePhase(float cosTheta, float g)
```

Comment:
Is this meant to be used now or added for future?

Reply:
Options for people to try different algorithms. Vanilla is using SchlickPhase but the others are more physically accurate (at a perf cost).